tests/integration/high_availability/test_self_healing.py

```diff
@@ -315,3 +319,99 @@ async def test_network_cut_affecting_an_instance(
     assert isolated_primary_memberrole == "secondary"
 
     await ensure_all_units_continuous_writes_incrementing(ops_test)
+
+
+@pytest.mark.order(2)
+@pytest.mark.abort_on_fail
+@pytest.mark.self_healing_tests
+async def test_graceful_full_cluster_crash_test(
+    ops_test: OpsTest, continuous_writes, restart_policy
+) -> None:
+    """Test to send SIGTERM to all units and then ensure that the cluster recovers."""
+    mysql_application_name, application_name = await high_availability_test_setup(ops_test)
+
+    time.sleep(30)
```

Comment:
@shayancanonical I'm having intermittent failures on continuous writes, and I wonder if this `sleep` is to address scenarios like that

Reply:
@paulomach this was added to mitigate against mysqld being restarted in the units after updating the pebble plan (to set `ignore` for `on-success` and `on-failure`). apparently, pebble plans of multiple units were being updated concurrently and this was leading to the cluster going into a funky state (coming back with GR disabled). after carefully updating pebble plans (one unit at a time), the need for this sleep statement has disappeared